backend/app/db/schemas/search.py

```diff
@@ -0,0 +1,115 @@
+from enum import Enum
+from typing import Dict, List, Optional, Tuple
+
+from pydantic import BaseModel
+
+
+class SortEnum(str, Enum):
+    """Sort ordering for use building OpenSearch query body."""
+
+    ASCENDING = "asc"
+    DESCENDING = "desc"
+
+
+class SortField(str, Enum):
+    """Sort field for use building OpenSearch query body."""
+
+    DATE = "date"
+    TITLE = "title"
+    TOP_HIT = "top_hit"
+    # TODO: complete enum
+
+
+class SearchRequestBody(BaseModel):
+    """The request body expected by the search API endpoint."""
+
+    query_string: str
+    exact_match: bool
+    max_passages_per_doc: int = 10  # TODO: decide on default
+
+    # TODO: Improve filters to allow generics & use filter types
+    keyword_filters: Optional[Dict[str, List[str]]] = None
+    year_range: Optional[Tuple[Optional[int], Optional[int]]] = None
+
+    sort_field: SortField = SortField.DATE
+    sort_order: SortEnum = SortEnum.DESCENDING
+
+    limit: int = 10  # TODO: decide on default
+    offset: int = 0
+
+
+class SearchResponseDocumentPassage(BaseModel):
+    """A Document passage match returned by the search API endpoint."""
+
+    text: str
+    text_block_id: str
+    text_block_page: int
+    text_block_coords: List[List[int]]
```

Comment:
I think the type of coords should be `List[List[float]]`

Reply:
I think I might go a bit further and actually define some types, but I'll make sure to define `Coord = Tuple[float, float`]